### ISSUE TITLE
AssertionError: weights with shape (22, 256) does not match shape=(21, 256) dtype=dtype('float32') -> Jax related?

### ISSUE THREAD

**Issue by clemensgrimm** [deepmind/alphafold#308]
Running command for oligotetramer sequence 

```
 python3 docker/run_docker.py  --fasta_paths=input.fas --max_template_date=2022-05-14  --data_dir=parameter/ --model_preset=multimer
```

crashes after jackhmm and hhblits:
```
 AssertionError: 'alphafold/alphafold_iteration/evoformer/preprocess_1d/weights' with shape (22, 256) does not match shape=(21, 256) dtype=dtype('float32')
```

Full output:
```
I1213 20:58:40.690539 140099439195072 run_docker.py:222] I1213 19:58:40.689060 140041849176640 pipeline.py:221] Uniref90 MSA size: 5649 sequences.
I1213 20:58:40.690940 140099439195072 run_docker.py:222] I1213 19:58:40.689277 140041849176640 pipeline.py:222] BFD MSA size: 1645 sequences.
I1213 20:58:40.691152 140099439195072 run_docker.py:222] I1213 19:58:40.689318 140041849176640 pipeline.py:223] MGnify MSA size: 501 sequences.
I1213 20:58:40.691330 140099439195072 run_docker.py:222] I1213 19:58:40.689357 140041849176640 pipeline.py:225] Final (deduplicated) MSA size: 7678 sequences.
I1213 20:58:40.691499 140099439195072 run_docker.py:222] I1213 19:58:40.689397 140041849176640 pipeline.py:228] Total number of templates (NB: this can include bad templates and is later filtered to top 4): 20.
I1213 20:58:40.701009 140099439195072 run_docker.py:222] I1213 19:58:40.700000 140041849176640 jackhmmer.py:130] Launching subprocess "/usr/bin/jackhmmer -o /dev/null -A /tmp/tmpan7g1yx4/output.sto --noali --F1 0.0005 --F2 5e-05 --F3 5e-07 --incE 0.0001 -E 0.0001 --cpu 8 -N 1 /tmp/tmpr0odyfcj.fasta /mnt/uniprot_database_path/uniprot.fasta"
I1213 20:58:40.930068 140099439195072 run_docker.py:222] I1213 19:58:40.928780 140041849176640 utils.py:36] Started Jackhmmer (uniprot.fasta) query
I1213 21:20:46.942055 140099439195072 run_docker.py:222] I1213 20:20:46.940270 140041849176640 utils.py:40] Finished Jackhmmer (uniprot.fasta) query in 1326.011 seconds
I1213 21:20:54.538525 140099439195072 run_docker.py:222] I1213 20:20:54.537352 140041849176640 run_alphafold.py:185] Running model model_1_multimer on Larp1-1433-FL
I1213 21:20:54.538951 140099439195072 run_docker.py:222] I1213 20:20:54.538107 140041849176640 model.py:166] Running predict with shape(feat) = {'aatype': (1296,), 'residue_index': (1296,), 'seq_length': (), 'msa': (3072, 1296), 'num_alignments': (), 'template_aatype': (4, 1296), 'template_all_atom_mask': (4, 1296, 37), 'template_all_atom_positions': (4, 1296, 37, 3), 'asym_id': (1296,), 'sym_id': (1296,), 'entity_id': (1296,), 'deletion_matrix': (3072, 1296), 'deletion_mean': (1296,), 'all_atom_mask': (1296, 37), 'all_atom_positions': (1296, 37, 3), 'assembly_num_chains': (), 'entity_mask': (1296,), 'num_templates': (), 'cluster_bias_mask': (3072,), 'bert_mask': (3072, 1296), 'seq_mask': (1296,), 'msa_mask': (3072, 1296)}
I1213 21:20:55.089957 140099439195072 run_docker.py:222] Traceback (most recent call last):
I1213 21:20:55.090378 140099439195072 run_docker.py:222] File "/app/alphafold/run_alphafold.py", line 427, in <module>
I1213 21:20:55.090594 140099439195072 run_docker.py:222] app.run(main)
I1213 21:20:55.090770 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 312, in run
I1213 21:20:55.090939 140099439195072 run_docker.py:222] _run_main(main, args)
I1213 21:20:55.091105 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 258, in _run_main
I1213 21:20:55.091457 140099439195072 run_docker.py:222] sys.exit(main(argv))
I1213 21:20:55.091624 140099439195072 run_docker.py:222] File "/app/alphafold/run_alphafold.py", line 412, in main
I1213 21:20:55.091785 140099439195072 run_docker.py:222] is_prokaryote=is_prokaryote)
I1213 21:20:55.091945 140099439195072 run_docker.py:222] File "/app/alphafold/run_alphafold.py", line 194, in predict_structure
I1213 21:20:55.092105 140099439195072 run_docker.py:222] random_seed=model_random_seed)
I1213 21:20:55.092265 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/model.py", line 167, in predict
I1213 21:20:55.092427 140099439195072 run_docker.py:222] result = self.apply(self.params, jax.random.PRNGKey(random_seed), feat)
I1213 21:20:55.092590 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
I1213 21:20:55.092752 140099439195072 run_docker.py:222] return fun(*args, **kwargs)
I1213 21:20:55.092911 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/api.py", line 427, in cache_miss
I1213 21:20:55.093070 140099439195072 run_docker.py:222] donated_invars=donated_invars, inline=inline)
I1213 21:20:55.093226 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 1661, in bind
I1213 21:20:55.093387 140099439195072 run_docker.py:222] return call_bind(self, fun, *args, **params)
I1213 21:20:55.093545 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 1652, in call_bind
I1213 21:20:55.093703 140099439195072 run_docker.py:222] outs = primitive.process(top_trace, fun, tracers, params)
I1213 21:20:55.093862 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 1664, in process
I1213 21:20:55.094020 140099439195072 run_docker.py:222] return trace.process_call(self, fun, tracers, params)
I1213 21:20:55.094178 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/core.py", line 633, in process_call
I1213 21:20:55.094377 140099439195072 run_docker.py:222] return primitive.impl(f, *tracers, **params)
I1213 21:20:55.094536 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/dispatch.py", line 129, in _xla_call_impl
I1213 21:20:55.094702 140099439195072 run_docker.py:222] *unsafe_map(arg_spec, args))
I1213 21:20:55.094774 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/linear_util.py", line 263, in memoized_fun
I1213 21:20:55.094846 140099439195072 run_docker.py:222] ans = call(fun, *args)
I1213 21:20:55.094919 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/dispatch.py", line 156, in _xla_callable_uncached
I1213 21:20:55.094993 140099439195072 run_docker.py:222] *arg_specs).compile().unsafe_call
I1213 21:20:55.095067 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/profiler.py", line 206, in wrapper
I1213 21:20:55.095139 140099439195072 run_docker.py:222] return func(*args, **kwargs)
I1213 21:20:55.095212 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/dispatch.py", line 170, in lower_xla_callable
I1213 21:20:55.095284 140099439195072 run_docker.py:222] fun, abstract_args, pe.debug_info_final(fun, "jit"))
I1213 21:20:55.095357 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/profiler.py", line 206, in wrapper
I1213 21:20:55.095429 140099439195072 run_docker.py:222] return func(*args, **kwargs)
I1213 21:20:55.095502 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 1566, in trace_to_jaxpr_final
I1213 21:20:55.095575 140099439195072 run_docker.py:222] jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(fun, main, in_avals)
I1213 21:20:55.095647 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 1543, in trace_to_subjaxpr_dynamic
I1213 21:20:55.095720 140099439195072 run_docker.py:222] ans = fun.call_wrapped(*in_tracers)
I1213 21:20:55.095793 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/linear_util.py", line 166, in call_wrapped
I1213 21:20:55.095867 140099439195072 run_docker.py:222] ans = self.f(*args, **dict(self.params, **kwargs))
I1213 21:20:55.095938 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/transform.py", line 125, in apply_fn
I1213 21:20:55.096012 140099439195072 run_docker.py:222] out, state = f.apply(params, {}, *args, **kwargs)
I1213 21:20:55.096085 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/transform.py", line 313, in apply_fn
I1213 21:20:55.096158 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.096231 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/model.py", line 79, in _forward_fn
I1213 21:20:55.096304 140099439195072 run_docker.py:222] is_training=False)
I1213 21:20:55.096377 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.096450 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.096522 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.096595 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.096667 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 484, in __call__
I1213 21:20:55.096739 140099439195072 run_docker.py:222] prev, safe_key = hk.fori_loop(0, num_iter, recycle_body, (prev, safe_key))
I1213 21:20:55.096811 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/stateful.py", line 544, in fori_loop
I1213 21:20:55.096884 140099439195072 run_docker.py:222] state, val = jax.lax.fori_loop(lower, upper, pure_body_fun, init_val)
I1213 21:20:55.096956 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
I1213 21:20:55.097029 140099439195072 run_docker.py:222] return fun(*args, **kwargs)
I1213 21:20:55.097102 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/lax/control_flow.py", line 222, in fori_loop
I1213 21:20:55.097174 140099439195072 run_docker.py:222] None, length=upper_ - lower_)
I1213 21:20:55.097246 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
I1213 21:20:55.097318 140099439195072 run_docker.py:222] return fun(*args, **kwargs)
I1213 21:20:55.097390 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/lax/control_flow.py", line 1505, in scan
I1213 21:20:55.097464 140099439195072 run_docker.py:222] init_flat, carry_avals, carry_avals_out, init_tree, *rest = _create_jaxpr(init)
I1213 21:20:55.097536 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/lax/control_flow.py", line 1492, in _create_jaxpr
I1213 21:20:55.097609 140099439195072 run_docker.py:222] f, in_tree, carry_avals + x_avals, "scan")
I1213 21:20:55.097682 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/util.py", line 202, in wrapper
I1213 21:20:55.097756 140099439195072 run_docker.py:222] return cached(config._trace_context(), *args, **kwargs)
I1213 21:20:55.097829 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/util.py", line 195, in cached
I1213 21:20:55.097901 140099439195072 run_docker.py:222] return f(*args, **kwargs)
I1213 21:20:55.097973 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/lax/control_flow.py", line 82, in _initial_style_jaxpr
I1213 21:20:55.098046 140099439195072 run_docker.py:222] fun, in_tree, in_avals, primitive_name)
I1213 21:20:55.098118 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/util.py", line 202, in wrapper
I1213 21:20:55.098191 140099439195072 run_docker.py:222] return cached(config._trace_context(), *args, **kwargs)
I1213 21:20:55.098269 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/util.py", line 195, in cached
I1213 21:20:55.098343 140099439195072 run_docker.py:222] return f(*args, **kwargs)
I1213 21:20:55.098416 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/lax/control_flow.py", line 75, in _initial_style_open_jaxpr
I1213 21:20:55.098490 140099439195072 run_docker.py:222] jaxpr, _, consts = pe.trace_to_jaxpr_dynamic(wrapped_fun, in_avals, debug)
I1213 21:20:55.098568 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/profiler.py", line 206, in wrapper
I1213 21:20:55.098641 140099439195072 run_docker.py:222] return func(*args, **kwargs)
I1213 21:20:55.098713 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 1533, in trace_to_jaxpr_dynamic
I1213 21:20:55.098785 140099439195072 run_docker.py:222] jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(fun, main, in_avals)
I1213 21:20:55.098858 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 1543, in trace_to_subjaxpr_dynamic
I1213 21:20:55.098932 140099439195072 run_docker.py:222] ans = fun.call_wrapped(*in_tracers)
I1213 21:20:55.099005 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/linear_util.py", line 166, in call_wrapped
I1213 21:20:55.099078 140099439195072 run_docker.py:222] ans = self.f(*args, **dict(self.params, **kwargs))
I1213 21:20:55.099151 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/lax/control_flow.py", line 148, in scanned_fun
I1213 21:20:55.099222 140099439195072 run_docker.py:222] return (i + 1, body_fun(i, x)), None
I1213 21:20:55.099294 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/stateful.py", line 524, in pure_body_fun
I1213 21:20:55.099366 140099439195072 run_docker.py:222] val = body_fun(i, val)
I1213 21:20:55.099439 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 481, in recycle_body
I1213 21:20:55.099511 140099439195072 run_docker.py:222] ret = apply_network(prev=prev, safe_key=safe_key2)
I1213 21:20:55.099585 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 452, in apply_network
I1213 21:20:55.099658 140099439195072 run_docker.py:222] safe_key=safe_key)
I1213 21:20:55.099731 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.099803 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.099875 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.099948 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.100020 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 322, in __call__
I1213 21:20:55.100093 140099439195072 run_docker.py:222] lambda: embedding_module(batch, is_training))
I1213 21:20:55.100166 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/stateful.py", line 665, in eval_shape
I1213 21:20:55.100321 140099439195072 run_docker.py:222] out_shape = jax.eval_shape(fun, *args, **kwargs)
I1213 21:20:55.100396 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/api.py", line 2956, in eval_shape
I1213 21:20:55.100470 140099439195072 run_docker.py:222] debug_info=debug_info)
I1213 21:20:55.100543 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 420, in abstract_eval_fun
I1213 21:20:55.100617 140099439195072 run_docker.py:222] lu.wrap_init(fun, params), avals, debug_info)
I1213 21:20:55.100690 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/_src/profiler.py", line 206, in wrapper
I1213 21:20:55.100761 140099439195072 run_docker.py:222] return func(*args, **kwargs)
I1213 21:20:55.100834 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 1533, in trace_to_jaxpr_dynamic
I1213 21:20:55.100906 140099439195072 run_docker.py:222] jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(fun, main, in_avals)
I1213 21:20:55.100978 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/interpreters/partial_eval.py", line 1543, in trace_to_subjaxpr_dynamic
I1213 21:20:55.101050 140099439195072 run_docker.py:222] ans = fun.call_wrapped(*in_tracers)
I1213 21:20:55.101122 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/linear_util.py", line 166, in call_wrapped
I1213 21:20:55.101193 140099439195072 run_docker.py:222] ans = self.f(*args, **dict(self.params, **kwargs))
I1213 21:20:55.101266 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/jax/linear_util.py", line 166, in call_wrapped
I1213 21:20:55.101338 140099439195072 run_docker.py:222] ans = self.f(*args, **dict(self.params, **kwargs))
I1213 21:20:55.101411 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 322, in <lambda>
I1213 21:20:55.101483 140099439195072 run_docker.py:222] lambda: embedding_module(batch, is_training))
I1213 21:20:55.101556 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.101629 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.101702 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.101775 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.101849 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 595, in __call__
I1213 21:20:55.101923 140099439195072 run_docker.py:222] target_feat)
I1213 21:20:55.101995 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.102067 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.102138 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.102211 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.102290 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/common_modules.py", line 118, in __call__
I1213 21:20:55.102362 140099439195072 run_docker.py:222] weight_init)
I1213 21:20:55.102434 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/base.py", line 319, in get_parameter
I1213 21:20:55.102505 140099439195072 run_docker.py:222] fq_name, param.shape, shape, dtype))
I1213 21:20:55.102579 140099439195072 run_docker.py:222] jax._src.traceback_util.UnfilteredStackTrace: AssertionError: 'alphafold/alphafold_iteration/evoformer/preprocess_1d/weights' with shape (22, 256) does not match shape=(21, 256) dtype=dtype('float32')
I1213 21:20:55.102655 140099439195072 run_docker.py:222]
I1213 21:20:55.102728 140099439195072 run_docker.py:222] The stack trace below excludes JAX-internal frames.
I1213 21:20:55.102801 140099439195072 run_docker.py:222] The preceding is the original exception that occurred, unmodified.
I1213 21:20:55.102873 140099439195072 run_docker.py:222]
I1213 21:20:55.102944 140099439195072 run_docker.py:222] --------------------
I1213 21:20:55.103016 140099439195072 run_docker.py:222]
I1213 21:20:55.103089 140099439195072 run_docker.py:222] The above exception was the direct cause of the following exception:
I1213 21:20:55.103161 140099439195072 run_docker.py:222]
I1213 21:20:55.103234 140099439195072 run_docker.py:222] Traceback (most recent call last):
I1213 21:20:55.103306 140099439195072 run_docker.py:222] File "/app/alphafold/run_alphafold.py", line 427, in <module>
I1213 21:20:55.103378 140099439195072 run_docker.py:222] app.run(main)
I1213 21:20:55.103450 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 312, in run
I1213 21:20:55.103521 140099439195072 run_docker.py:222] _run_main(main, args)
I1213 21:20:55.103592 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 258, in _run_main
I1213 21:20:55.103665 140099439195072 run_docker.py:222] sys.exit(main(argv))
I1213 21:20:55.103737 140099439195072 run_docker.py:222] File "/app/alphafold/run_alphafold.py", line 412, in main
I1213 21:20:55.103809 140099439195072 run_docker.py:222] is_prokaryote=is_prokaryote)
I1213 21:20:55.103882 140099439195072 run_docker.py:222] File "/app/alphafold/run_alphafold.py", line 194, in predict_structure
I1213 21:20:55.103955 140099439195072 run_docker.py:222] random_seed=model_random_seed)
I1213 21:20:55.104027 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/model.py", line 167, in predict
I1213 21:20:55.104099 140099439195072 run_docker.py:222] result = self.apply(self.params, jax.random.PRNGKey(random_seed), feat)
I1213 21:20:55.104171 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/transform.py", line 125, in apply_fn
I1213 21:20:55.104243 140099439195072 run_docker.py:222] out, state = f.apply(params, {}, *args, **kwargs)
I1213 21:20:55.104316 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/transform.py", line 313, in apply_fn
I1213 21:20:55.104389 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.104462 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/model.py", line 79, in _forward_fn
I1213 21:20:55.104534 140099439195072 run_docker.py:222] is_training=False)
I1213 21:20:55.104606 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.104679 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.104751 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.104822 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.104894 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 484, in __call__
I1213 21:20:55.104967 140099439195072 run_docker.py:222] prev, safe_key = hk.fori_loop(0, num_iter, recycle_body, (prev, safe_key))
I1213 21:20:55.105039 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/stateful.py", line 544, in fori_loop
I1213 21:20:55.105110 140099439195072 run_docker.py:222] state, val = jax.lax.fori_loop(lower, upper, pure_body_fun, init_val)
I1213 21:20:55.105181 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/stateful.py", line 524, in pure_body_fun
I1213 21:20:55.105254 140099439195072 run_docker.py:222] val = body_fun(i, val)
I1213 21:20:55.105325 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 481, in recycle_body
I1213 21:20:55.105397 140099439195072 run_docker.py:222] ret = apply_network(prev=prev, safe_key=safe_key2)
I1213 21:20:55.105469 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 452, in apply_network
I1213 21:20:55.105541 140099439195072 run_docker.py:222] safe_key=safe_key)
I1213 21:20:55.105615 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.105688 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.105761 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.105834 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.105907 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 322, in __call__
I1213 21:20:55.105979 140099439195072 run_docker.py:222] lambda: embedding_module(batch, is_training))
I1213 21:20:55.106050 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/stateful.py", line 665, in eval_shape
I1213 21:20:55.106122 140099439195072 run_docker.py:222] out_shape = jax.eval_shape(fun, *args, **kwargs)
I1213 21:20:55.106194 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 322, in <lambda>
I1213 21:20:55.106284 140099439195072 run_docker.py:222] lambda: embedding_module(batch, is_training))
I1213 21:20:55.106357 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.106430 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.106503 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.106575 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.106647 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/modules_multimer.py", line 595, in __call__
I1213 21:20:55.106719 140099439195072 run_docker.py:222] target_feat)
I1213 21:20:55.106791 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 428, in wrapped
I1213 21:20:55.106863 140099439195072 run_docker.py:222] out = f(*args, **kwargs)
I1213 21:20:55.106935 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/module.py", line 279, in run_interceptors
I1213 21:20:55.107008 140099439195072 run_docker.py:222] return bound_method(*args, **kwargs)
I1213 21:20:55.107081 140099439195072 run_docker.py:222] File "/app/alphafold/alphafold/model/common_modules.py", line 118, in __call__
I1213 21:20:55.107153 140099439195072 run_docker.py:222] weight_init)
I1213 21:20:55.107225 140099439195072 run_docker.py:222] File "/opt/conda/lib/python3.7/site-packages/haiku/_src/base.py", line 319, in get_parameter
I1213 21:20:55.107298 140099439195072 run_docker.py:222] fq_name, param.shape, shape, dtype))
I1213 21:20:55.107369 140099439195072 run_docker.py:222] AssertionError: 'alphafold/alphafold_iteration/evoformer/preprocess_1d/weights' with shape (22, 256) does not match shape=(21, 256) dtype=dtype('float32')
```

**Comment by clemensgrimm** [comment]
Seems to be a result of the fix proposed in #298 which results in a broken multimer installation. No apparent issues when using monomer mode.

**Comment by sami-chaaban** [comment]
I'm still geting this error. How did you solve it?
